src/shared/defaultAwsContextCommands.ts

```diff
@@ -87,13 +87,13 @@ export class DefaultAWSContextCommands {
             if (successfulLogin) {
                 this.refresh()
                 await this.checkExplorerForDefaultRegion(profileName)
+            } else {
+                await this.onCommandLogout()
+                // tslint:disable-next-line: no-floating-promises
+                UserCredentialsUtils.badUserCredentialPrompt(profileName)
             }
         }
-        if (!profileName || !successfulLogin) {
-            // credentials are invalid. Prompt user and log out
-            // TODO: Prompt user!
-            await this.onCommandLogout()
-        }
+        // do nothing if !profileName: that means user didn't select from the quick pick.
```

Comment:
If we explicitly exit after line 83, the rest of the function is dedicated to dealing with logging in

Reply:
✔️